Source/WebCore/css/parser/CSSParserFastPaths.cpp

```diff
@@ -611,18 +611,30 @@ static std::optional<SRGBA<uint8_t>> parseNumericColor(std::span<const Character
             return *hexColor;
     }
 
-    // FIXME: rgb() and rgba() are now synonyms, so we should collapse these two clauses together. webkit.org/b/276761
-    if (mightBeRGBA(characters)) {
+    if (mightBeRGB(characters) || mightBeRGBA(characters)) {
         auto expectedUnitType = CSSUnitType::CSS_UNKNOWN;
+        auto current = mightBeRGBA(characters) ? characters.subspan(5) : characters.subspan(4); // handle 'a' for 'rgba()' case
 
-        auto current = characters.subspan(5);
+        // red and green will both terminate with ','
         auto red = parseColorIntOrPercentage(current, ',', expectedUnitType);
         if (!red)
             return std::nullopt;
         auto green = parseColorIntOrPercentage(current, ',', expectedUnitType);
         if (!green)
             return std::nullopt;
-        auto blue = parseColorIntOrPercentage(current, ',', expectedUnitType);
+
+        // check if blue terminates with ')' (no alpha-value provided)
+        auto backup = current; // current parsing implementation consumes span, so back up in case first parse fails
+        auto blue = parseColorIntOrPercentage(current, ')', expectedUnitType);
```

Comment:
It' s a bit sad that we try to parse the blue part twice in case of rgba (especially in a fast path function). 
Maybe dispatch the blue parsing based on `mightBeRGBA()` (and it would help to remove the error prone manipulation of `backup`).
Also, maybe the compiler would optimize it anyway, but using a constant to store the result of `mightBeRGB(A)`  could help.

Reply:
Seems that `rgb()` and `rgba()` parse exactly the same syntax actually so we can ignore my comment about dispatch based on `mightBeRGBA()` 😬  https://drafts.csswg.org/css-color/#rgb-functions
So we should just avoid parsing twice the integer number just because of a different terminator. Using a `std::optional<char> terminator` and doing the terminator parsing manually could work.